human_general_analysis/nbconverted/Try_other_enrichment_methods.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Name
+# 
+# This notebook plugs in other gene set enrichment methods to demonstrate that our method, SOPHIE, can be inserted into different pipelines and work with other methods
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('load_ext', 'rpy2.ipython')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import sys
+import pandas as pd
+import numpy as np
+import pickle
+
+from rpy2.robjects import pandas2ri
+pandas2ri.activate()
+
+from ponyo import utils
+from generic_expression_patterns_modules import calc, process
+
+np.random.seed(123)
+
+
+# In[2]:
+
+
+# Read in config variables
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
```

Comment:
Might want to try and use pathlib. I know broken record, but it definitely makes os path generation a lot easier to deal with.

Reply:
Not a broken record. I definitely agree with using paths instead of the way I currently have my config. The legacy reason for having the config this way is 1) to be consistent with how ponyo is, which this repo uses and 2) it seemed more user friendly at the time to have a tab-delimited file and have that passed into the notebook as opposed to "importing a python file. I also wasn't aware, at the time when I created the config, that I could define both variables and paths in the same file. But rest assured that my other [newer repo](https://github.com/greenelab/core-accessory-interactome) uses paths:)